stan/math/rev/functor/algebra_solver_powell.hpp

```diff
@@ -21,57 +22,77 @@ namespace math {
  * function theorem. The call to Jacobian() occurs outside the call to
  * chain() -- this prevents malloc issues.
  */
-template <typename Fs, typename F, typename T, typename Fx>
+template <typename Fy, typename T, typename Fx>
 struct algebra_solver_vari : public vari {
-  /** vector of parameters */
-  vari** y_;
   /** number of parameters */
-  int y_size_;
+  const int y_size_;
+  /** value of parameters */
+  const Eigen::Matrix<T, Eigen::Dynamic, 1> y_val_;
+  /** System functor (f_) w.r.t. inputs (y_) */
+  Fy fy_;
+  /** array of parameters */
+  vari** y_;
   /** number of unknowns */
-  int x_size_;
-  /** vector of solution */
-  vari** theta_;
-  /** Jacobian of the solution w.r.t parameters */
-  double* Jx_y_;
+  const int x_size_;
+  /** value of unknowns */
+  const Eigen::VectorXd& x_val_;
+  /** Hybrj functor solver (f_) w.r.t. outputs (x_) */
+  Fx fx_;
+  /** array of unknowns */
+  vari** x_;
+  /** Jacobian of f w.r.t. outputs (x_) */
+  const Eigen::MatrixXd Jf_x_;
```

Comment:
```suggestion
arena_t<Eigen::MatrixXd> Jf_x_;
```

Reply:
In general I'd remove const from any class members

Reply:
Sounds good! (I'll resolve this for now, but I'm curious why that's the case. I'm reasonably new to `C++`, but I was under the impression that, in general, if something can be `const` it's good practice to make it `const`. Is the issue here that actually these members might need to be changed for some reason that I'm not seeing, or is there a different concern?)